sycl/test-e2e/bindless_images/vulkan_interop/sampled_images.cpp

```diff
@@ -104,8 +129,14 @@ bool run_sycl(InteropHandleT inputInteropMemHandle,
   using OutType = typename OutputType<DType, CType>::type;
   using VecType = sycl::vec<OutType, NChannels>;
 
-  auto handles = create_test_handles(ctxt, dev, samp, inputInteropMemHandle,
-                                     desc, img_size);
+  auto handles =
+      create_test_handles(ctxt, dev, samp, inputInteropMemHandle,
+                          sycl_wait_semaphore_handle, desc, img_size);
+
+#ifdef TEST_SEMAPHORE_IMPORT
+  // Extension: wait for imported semaphore
+  q.ext_oneapi_wait_external_semaphore(handles.sycl_wait_external_semaphore);
```

Comment:
so to clarify, the sequence is:
1. submit VK operations and signal the semaphore
2. SYCL waits on `sycl_wait_external_semaphore` to ensure VK operations are complete before queuing up any compute operation
3. q.wait_and_throw() is simply to wait on the sycl Q for compute to complete right? we don't have any signal on an external semaphore that VK will wait on here?

Reply:
in the unsampled test case we have a `sycl_done_semaphore` that is passed into q.submit() but Im not sure who is waiting on that: https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/bindless_images/vulkan_interop/unsampled_images.cpp#L251

Reply:
> so to clarify, the sequence is:
> 
> 1. submit VK operations and signal the semaphore
> 2. SYCL waits on `sycl_wait_external_semaphore` to ensure VK operations are complete before queuing up any compute operation
> 3. q.wait_and_throw() is simply to wait on the sycl Q for compute to complete right? we don't have any signal on an external semaphore that VK will wait on here?

yes, correct.

Reply:
> in the unsampled test case we have a `sycl_done_semaphore` that is passed into q.submit() but Im not sure who is waiting on that: https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/bindless_images/vulkan_interop/unsampled_images.cpp#L251

in that test vulkan is waiting on the semaphore at https://github.com/intel/llvm/blob/6a9de65917078f0cb40553250d63ab5993d25c49/sycl/test-e2e/bindless_images/vulkan_interop/unsampled_images.cpp#L527 , i.e. right before reading vulkan image for verification.

This test doesn't have this workflow because as explained in the commit message this test uses sycl buffer as output and there is no interop support between sycl buffer and vulkan buffer.